tests/test_analysis.py

```diff
@@ -0,0 +1,791 @@
+            self.gene_rank_records,
+            {
+                1: {
+                    "Phenopacket": "phenopacket.json",
+                    "Gene": "PLXNA1",
+                    Path("/path/to/results_dir"): 0,
+                },
+                2: {
+                    "Phenopacket": "phenopacket.json",
+                    "Gene": "LARGE1",
+                    Path("/path/to/results_dir"): 0,
+                },
+            },
+        )
+
+    def test_assess_gene_prioritisation_threshold_meets_cutoff(self):
+        assess_with_threshold = copy(self.assess_gene_prioritisation)
+        assess_with_threshold.threshold = 0.1
+        assess_with_threshold.assess_gene_prioritisation(
+            self.gene_rank_stats, self.gene_rank_records
+        )
+        self.assertEqual(
+            self.gene_rank_stats,
+            RankStats(top=1, top3=1, top5=1, found=1, total=2, reciprocal_ranks=[1.0]),
+        )
+        self.assertEqual(
+            self.gene_rank_records,
+            {
+                1: {
+                    "Phenopacket": "phenopacket.json",
+                    "Gene": "PLXNA1",
+                    Path("/path/to/results_dir"): 1,
+                },
+                2: {
+                    "Phenopacket": "phenopacket.json",
+                    "Gene": "LARGE1",
+                    Path("/path/to/results_dir"): 0,
+                },
+            },
+        )
+
+
+class TestAssessVariantPrioritisation(unittest.TestCase):
+    def setUp(self) -> None:
+        variant_results = pd.DataFrame(
+            [
+                {
+                    "variant": {
+                        "chrom": "3",
+                        "pos": 126730873,
+                        "ref": "G",
+                        "alt": "A",
+                        "gene": "PLXNA1",
+                    },
+                    "score": 0.0484,
+                    "rank": 1,
+                },
+                {
+                    "variant": {
+                        "chrom": "3",
+                        "pos": 126730873,
+                        "ref": "G",
+                        "alt": "A",
+                        "gene": "PLXNA1",
+                    },
+                    "score": 0.0484,
+                    "rank": 1,
+                },
+                {
+                    "variant": {
+                        "chrom": "3",
+                        "pos": 126741108,
+                        "ref": "G",
+                        "alt": "A",
+                        "gene": "PLXNA1",
+                    },
+                    "score": 0.0484,
+                    "rank": 1,
+                },
+            ]
+        )
+        variant_result_format = variant_results.drop("variant", axis=1).join(
+            variant_results.variant.apply(pd.Series)
+        )
+
+        self.assess_variant_prioritisation = AssessVariantPrioritisation(
+            phenopacket_path=Path("/path/to/phenopacket.json"),
```

Comment:
Just out of curiosity, why the need for adding this fake path here?

Reply:
The idea is that the path of the Phenopacket is recorded along with the gene/variant and rank so that this can produce tsv output files with the rank comparison across runs:

```
  | Phenopacket | Gene | /data/WHRI-Phenogenomics/projects/exomiser-benchmark/exomiser-cli-14.0.0-SNAPSHOT/exomiser_14_results/ | /data/WHRI-Phenogenomics/projects/exomiser-benchmark/exomiser-cli-13.1.0/exomiser_13.1_results/ | absolute_rank_difference
-- | -- | -- | -- | -- | --
1 | Abdul_Wahab-2016-GCDH-Patient_5-exomiser-pheval-gene-output.tsv | GCDH | 1 | 1 | 0
2 | Ajmal-2013-BBS1-IV-5_family_A-exomiser-pheval-gene-output.tsv | BBS1 | 1 | 1 | 0
3 | Al-Dosari-2010-TFAP2A-10-year-old_girl-exomiser-pheval-gene-output.tsv | TFAP2A | 2 | 2 | 0
```

This was done with the hopes of providing more detail into cases where the cases and ranks are greatly improving or on the other hand worsening. The fake path is needed there because the class takes in the Phenopacket path as an argument as the 'recording' of these ranks for each gene/variant for a proband is done in the methods under the class.

Reply:
Ok thank you! Makes sense.